htdocs/core/class/stats.class.php

```diff
@@ -318,7 +318,7 @@ public function getAllByProductEntry($year, $cachedelay = 0, $limit = 10)
 			dol_syslog(get_class($this).'::'.__FUNCTION__." read data from cache file ".$newpathofdestfile." ".$filedate.".");
 			$data = json_decode(file_get_contents($newpathofdestfile), true);
 		} else {
-			$data = $this->getAllByProduct($year, $limit);
+			$data = $this->getAllByProductEntry($year, $limit);
```

Comment:
You are inside the method getAllByProductEntry. If you call getAllByProductEntry here, you generate an infinte loop.